django_postmark_utils/admin.py

```diff
@@ -1,3 +1,149 @@
+from email.utils import formatdate
+import pickle
+
+from django.conf import settings
 from django.contrib import admin
+from django.core.mail.message import make_msgid
+from django.core.mail.utils import DNS_NAME
+from django.urls import reverse
+from django.utils.html import format_html
+from django.utils.translation import ugettext_lazy as _
+
+from .models import Bounce, Delivery, Message
+from .utils import ResendEmailMessage
+
+
+@admin.register(Message)
+class MessageAdmin(admin.ModelAdmin):
+
+    list_display = (
+        'message_id',
+        'date',
+        'subject',
+        'to_emails',
+        'delivery_status',
+        'delivery_message_id',
+        'delivery_error_code',
+    )
+    list_filter = (
+        'delivery_status',
+        'delivery_error_code',
+    )
+    search_fields = (
+        'message_id',
+        'to_emails',
+        'cc_emails',
+        'bcc_emails',
+        'delivery_message_id',
+    )
+
+
+def resend_messages(modeladmin, request, queryset):
+
+    message_bounces = {}
+    for obj in queryset:
+        message = obj.message
+        if message not in message_bounces:
+            message_bounces[message] = []
+        message_bounces[message].append(obj)
+    for message in message_bounces:
+        # Recreate the message
+        msg = pickle.loads(message.pickled_obj)
+
+        # TODO: Check if resent header fields can be used, and if so, do not
```

Comment:
I don't think we should use the original fields - for tracking purposes this should be a new message with a new ID.